pkgs/applications/audio/open-stage-control/default.nix

```diff
@@ -0,0 +1,83 @@
+{ pkgs, stdenv, lib, fetchFromGitHub, makeWrapper, makeDesktopItem, copyDesktopItems, nodejs, electron, python3, runCommand, open-stage-control, ... }:
+
+let
+  nodeComposition = import ./node-composition.nix {
+    inherit pkgs nodejs;
+    inherit (stdenv.hostPlatform) system;
+  };
+in
+
+nodeComposition.package.override rec {
+  pname = "open-stage-control";
+  inherit (nodeComposition.args) version;
+
+  src = fetchFromGitHub {
+    owner = "jean-emmanuel";
+    repo = "open-stage-control";
+    rev = "v${version}";
+    hash = "sha256-q18pRtsHfme+OPmi3LhJDK1AdpfkwhoE9LA2rNenDtY=";
+  };
+
+  nativeBuildInputs = [
+    copyDesktopItems
+    makeWrapper
+  ];
+
+  buildInputs = [
+    python3.pkgs.python-rtmidi
+  ];
+
+  dontNpmInstall = true;
+
+  preRebuild = ''
+    # remove electron dependencies from package.json
+    mv package.json package.orig.json
+    grep -v '"electron"\|"electron-installer-debian"\|"electron-packager"\|"electron-packager-plugin-non-proprietary-codecs-ffmpeg"' package.orig.json >package.json
+  '';
+
+  postInstall = ''
+    # fix shebangs in node_modules
+    patchShebangs --build $out/lib/node_modules/open-stage-control/node_modules/
+
+    # build assets
+    npm run build
+
+    # copy icon
+    install -Dm644 resources/images/logo.png $out/share/icons/hicolor/256x256/apps/open-stage-control.png
+    install -Dm644 resources/images/logo.svg $out/share/icons/hicolor/scalable/apps/open-stage-control.svg
+
+    # wrap electron and include python-rtmidi
+    makeWrapper '${electron}/bin/electron' $out/bin/open-stage-control \
+      --argv0 $out/bin/open-stage-control \
```

Comment:
It looks like if you exclude `--argv0`, it will just not specify `-a` on the `exec` command (see [here](https://github.com/NixOS/nixpkgs/blob/03f42a04ffc513c56e54d714c44414f091a0ab22/pkgs/build-support/setup-hooks/make-wrapper.sh#L185)), so argv0 would be `${electron}/bin/electron` rather than `$out/bin/open-stage-control`. I'm using `makeWrapper` here instead of `wrapProgram` because it is wrapping across derivations, but let me know if there is a better way I should be doing this

I just pushed an update that uses `makeBinaryWrapper` and `--inherit-argv0` which is a bit nicer too

Thank you again for the review!